crates/bevy_time/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+mod fixed_timestep;
+mod stopwatch;
+#[allow(clippy::module_inception)]
+mod time;
+mod timer;
+
+pub use fixed_timestep::*;
+pub use stopwatch::*;
+pub use time::*;
+pub use timer::*;
+
+pub mod prelude {
+    //! The Bevy Time Prelude.
+    #[doc(hidden)]
+    pub use crate::{Time, Timer};
+}
+
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+
+/// Adds time functionality to Apps.
+#[derive(Default)]
+pub struct TimePlugin;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash, SystemLabel)]
+/// Updates the elapsed time. Any system that interacts with [Time] component should run after
+/// this.
+pub struct TimeSystem;
+
+impl Plugin for TimePlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<Time>()
+            .init_resource::<FixedTimesteps>()
+            .register_type::<Timer>()
+            // time system is added as an "exclusive system" to ensure it runs before other systems
+            // in CoreStage::First
+            .add_system_to_stage(
+                CoreStage::First,
+                time_system.exclusive_system().label(TimeSystem),
```

Comment:
```suggestion
                time_system.exclusive_system().at_start().label(TimeSystem),
```

This fixes the timing: by default exclusive systems run at the end of the stage, not the beginning.

Reply:
Did this change since bevy_core was first written? Because this is just verbatim what was there previously.

Reply:
> by default exclusive systems run at the end of the stage, not the beginning

This is not correct. `AtStart` is the [default insertion point][1], so adding the specifier isn't necessary.

[1]: https://github.com/bevyengine/bevy/blob/0e821da7045852ff677c2b8ee945dc85a899b181/crates/bevy_ecs/src/schedule/system_descriptor.rs#L241-L251